hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieTableServiceClient.java

```diff
@@ -707,20 +709,34 @@ protected List<String> getInstantsToRollback(HoodieTableMetaClient metaClient, H
         }
       }).collect(Collectors.toList());
     } else if (cleaningPolicy.isLazy()) {
-      return inflightInstantsStream.filter(instant -> {
-        try {
-          return heartbeatClient.isHeartbeatExpired(instant.getTimestamp());
-        } catch (IOException io) {
-          throw new HoodieException("Failed to check heartbeat for instant " + instant, io);
-        }
-      }).map(HoodieInstant::getTimestamp).collect(Collectors.toList());
+      return getInstantsToRollbackForLazyCleanPolicy(metaClient, inflightInstantsStream);
     } else if (cleaningPolicy.isNever()) {
       return Collections.emptyList();
     } else {
       throw new IllegalArgumentException("Invalid Failed Writes Cleaning Policy " + config.getFailedWritesCleanPolicy());
     }
   }
 
+  @VisibleForTesting
```

Comment:
is it not possible to test it at the write client layer. would like to avoid using "VisibleForTesting" as much as possible.

Reply:
I move the test to `TestHoodieClientMultiWriter` and now still need to change the access modifier to protected. It's a little bit hard to test through calling `BaseHoodieTableServiceClient.rollbackFailedWrites` directly because the execution sequence is hard to control. 

So now the test will still get inflight instants before commit and then call `getInstantsToRollbackForLazyCleanPolicy` to check.